crates/core/src/kvs/index.rs

```diff
@@ -423,9 +554,27 @@ impl Building {
 				tx.del(ip).await?;
 
 				*count += 1;
-				self.set_status(BuildingStatus::UpdatesIndexing(*count)).await;
+				self.set_status(BuildingStatus::Indexing {
+					initial: Some(initial),
+					pending: Some(self.queue.read().await.pending() as usize),
+					updated: Some(*count),
+				})
+				.await;
 			}
 		}
 		Ok(())
 	}
+
+	fn abort(&self) {
+		self.aborted.store(true, Ordering::Relaxed);
+	}
+
+	async fn is_aborted(&self) -> bool {
+		if self.aborted.load(Ordering::Relaxed) {
```

Comment:
Might be a good idea to document the reasoning behind the ordering here. Why not `Acquire` or `SeqCst`?

Reply:
Done!